<<<<<<< HEAD:Waterfall-Proxy-Patches/0027-Always-release-packet.patch
From 7e00a0c75cc58bd8e69c2277b0d9b51986d52b81 Mon Sep 17 00:00:00 2001
=======
From 863b4ad3004b71c19707a0921dcaf2695b1e3247 Mon Sep 17 00:00:00 2001
>>>>>>> upstream/master:Waterfall-Proxy-Patches/0026-Always-release-packet.patch
From: linsaftw <25271111+linsaftw@users.noreply.github.com>
Date: Sat, 1 May 2021 23:28:19 -0300
Subject: [PATCH] Always release packet


diff --git a/proxy/src/main/java/net/md_5/bungee/netty/HandlerBoss.java b/proxy/src/main/java/net/md_5/bungee/netty/HandlerBoss.java
index e5602230..085cda11 100644
--- a/proxy/src/main/java/net/md_5/bungee/netty/HandlerBoss.java
+++ b/proxy/src/main/java/net/md_5/bungee/netty/HandlerBoss.java
@@ -113,12 +113,14 @@ public class HandlerBoss extends ChannelInboundHandlerAdapter
             return;
         }
 
-        if ( handler != null )
+        PacketWrapper packet = (PacketWrapper) msg;
+
+
+        try
         {
-            PacketWrapper packet = (PacketWrapper) msg;
-            boolean sendPacket = handler.shouldHandle( packet );
-            try
+            if ( handler != null )
             {
+                boolean sendPacket = handler.shouldHandle( packet );
                 if ( sendPacket && packet.packet != null )
                 {
                     try
@@ -133,10 +135,10 @@ public class HandlerBoss extends ChannelInboundHandlerAdapter
                 {
                     handler.handle( packet );
                 }
-            } finally
-            {
-                packet.trySingleRelease();
             }
+        } finally
+        {
+            packet.trySingleRelease();
         }
     }
 
-- 
2.32.0.windows.1

